examples/chef/common/clusters/channel/ChannelManager.cpp

```diff
@@ -0,0 +1,343 @@
+
+    ProgramType program1;
+    program1.identifier = chip::CharSpan::fromCharString("progid-abc1");
+    program1.channel    = abc;
+    program1.title      = chip::CharSpan::fromCharString("ABC Title1");
+    program1.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle1"));
+    program1.startTime  = 0;
+    program1.endTime    = 30 * 60;
+
+    mPrograms.push_back(program1);
+
+    ProgramType program_pbs1;
+    program_pbs1.identifier = chip::CharSpan::fromCharString("progid-pbs1");
+    program_pbs1.channel    = pbs;
+    program_pbs1.title      = chip::CharSpan::fromCharString("PBS Title1");
+    program_pbs1.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle1"));
+    program_pbs1.startTime  = 0;
+    program_pbs1.endTime    = 30 * 60;
+
+    mPrograms.push_back(program_pbs1);
+
+    ProgramType program2;
+    program2.identifier = chip::CharSpan::fromCharString("progid-abc2");
+    program2.channel    = abc;
+    program2.title      = chip::CharSpan::fromCharString("My Program Title2");
+    program2.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle2"));
+    program2.startTime  = 30 * 60;
+    program2.endTime    = 60 * 60;
+
+    mPrograms.push_back(program2);
+
+    ProgramType program3;
+    program3.identifier = chip::CharSpan::fromCharString("progid-abc3");
+    program3.channel    = abc;
+    program3.title      = chip::CharSpan::fromCharString("My Program Title3");
+    program3.subtitle   = MakeOptional(chip::CharSpan::fromCharString("My Program Subtitle3"));
+    program3.startTime  = 0;
+    program3.endTime    = 60 * 60;
+
+    mPrograms.push_back(program3);
+}
+
+CHIP_ERROR ChannelManager::HandleGetChannelList(AttributeValueEncoder & aEncoder)
+{
+    // TODO: Insert code here
+    return aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+        for (auto const & channel : ChannelManager().mChannels)
+        {
+            ReturnErrorOnFailure(encoder.Encode(channel));
+        }
+        return CHIP_NO_ERROR;
+    });
+}
+
+CHIP_ERROR ChannelManager::HandleGetLineup(AttributeValueEncoder & aEncoder)
+{
+    LineupInfoType lineup;
+    lineup.operatorName   = chip::CharSpan::fromCharString("Comcast");
+    lineup.lineupName     = MakeOptional(chip::CharSpan::fromCharString("Comcast King County"));
+    lineup.postalCode     = MakeOptional(chip::CharSpan::fromCharString("98052"));
+    lineup.lineupInfoType = chip::app::Clusters::Channel::LineupInfoTypeEnum::kMso;
+
+    return aEncoder.Encode(lineup);
+}
+
+CHIP_ERROR ChannelManager::HandleGetCurrentChannel(AttributeValueEncoder & aEncoder)
+{
+    return aEncoder.Encode(mCurrentChannel);
+}
+
+bool isChannelMatched(const ChannelInfoType & channel, const CharSpan & match)
+{
+    char number[32];
+    sprintf(number, "%d.%d", channel.majorNumber, channel.minorNumber);
+    bool nameMatch = channel.name.HasValue() ? channel.name.Value().data_equal(match) : false;
```

Comment:
instead of `foo ? bar : false` you can use `foo && bar`

Reply:
This whole thing could be:

```cpp
auto isMatch = [&match](const Optional<chip::CharSpan> &a)
{
    return a.HasValue() && a.Value().data_equal(match);
};

return isMatch(channel.name) 
    || isMatch(channel.affiliateCallSign) 
    || isMatch(channel.callSign)
    || match.data_equal(chip::CharSpan::fromCharString(number));
 ```
 
 not sure if more readable, howver it seems to have less complexity.

Reply:
It seems good. Done.